src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/AbstractImportCompletionProvider.cs

```diff
@@ -0,0 +1,230 @@
+
+                // Don't trigger import completion if the option value is "default" and the experiment is disabled for the user. 
+                if (importCompletionOptionValue == false ||
+                    (importCompletionOptionValue == null && !IsExperimentEnabled(document.Project.Solution.Workspace)))
+                {
+                    return;
+                }
+            }
+
+            using (Logger.LogBlock(FunctionId.Completion_TypeImportCompletionProvider_GetCompletionItemsAsync, cancellationToken))
+            {
+                // Find all namespaces in scope at current cursor location, 
+                // which will be used to filter so the provider only returns out-of-scope types.
+                var namespacesInScope = GetNamespacesInScope(document, syntaxContext, cancellationToken);
+                await AddCompletionItemsAsync(completionContext, syntaxContext, namespacesInScope, isExpandedCompletion, cancellationToken).ConfigureAwait(false);
+            }
+        }
+
+        private HashSet<string> GetNamespacesInScope(Document document, SyntaxContext syntaxContext, CancellationToken cancellationToken)
+        {
+            var semanticModel = syntaxContext.SemanticModel;
+            var importedNamespaces = GetImportedNamespaces(syntaxContext.LeftToken.Parent, semanticModel, cancellationToken);
+
+            // This hashset will be used to match namespace names, so it must have the same case-sensitivity as the source language.
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>()!;
+            var namespacesInScope = new HashSet<string>(importedNamespaces, syntaxFacts.StringComparer);
+
+            // Get containing namespaces.
+            var namespaceSymbol = semanticModel.GetEnclosingNamespace(syntaxContext.Position, cancellationToken);
+            while (namespaceSymbol != null)
+            {
+                namespacesInScope.Add(namespaceSymbol.ToDisplayString(SymbolDisplayFormats.NameFormat));
+                namespaceSymbol = namespaceSymbol.ContainingNamespace;
+            }
+
+            return namespacesInScope;
+        }
+
+        internal override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem completionItem, TextSpan completionListSpan, char? commitKey, CancellationToken cancellationToken)
+        {
+            var containingNamespace = ImportCompletionItem.GetContainingNamespace(completionItem);
+
+            if (await ShouldCompleteWithFullyQualifyTypeName().ConfigureAwait(false))
+            {
+                var fullyQualifiedName = $"{containingNamespace}.{completionItem.DisplayText}";
+                var change = new TextChange(completionListSpan, fullyQualifiedName);
+
+                return CompletionChange.Create(change);
+            }
+            else
```

Comment:
can you remove this else block and just dedent?

Reply:
Sure, fixed